core/src/main/scala/slog/LogBuilder.scala

```diff
@@ -0,0 +1,7 @@
+package slog
+
+trait LogBuilder[F[_]] {
+  def msg(msg: String): F[Unit]
+  def msg(ex: Throwable)(msg: String): F[Unit]
+  def context[T: StructureEncoder](key: String, value: => T): LogBuilder[F]
```

Comment:
this could come handy too
```scala
def context[T: StructureEncoder](m: Map[String, T]): LogBuilder[F]
```
or even better something kinda like
```scala
def context[T: StructureEncoder](m: Map[String, Lazy[T]]): LogBuilder[F]
```
where I don't yet know where to take `Lazy` from :)

Reply:
I don't think it's as useful as you think. Reason for that is you usually want to use different type for each key in the map.

Reply:
I think it would be used rarely, but when that time comes... but this is a minor discussion point anyway

Reply:
if that time comes, you can move the entire map into `T`, no?

Reply:
when I think of it, the method accepting `Map[String, T]` would rise in significance if we decided not to use the fluent/builder pattern and used flat methods instead

Reply:
@tomasherman but then it would be nested one level lower, which might not be what you want. but it's a detail now

Reply:
We could (once again :-) ) also provide implicit conversions from `StructureEncoder[T]` to `StructureEncoder[Map[String, T]]`. Regarding nesting, I think most would not care much.